src/backend/distributed/sql/udfs/citus_drop_trigger/latest.sql

```diff
@@ -34,6 +35,25 @@ BEGIN
         PERFORM master_remove_partition_metadata(v_obj.objid, v_obj.schema_name, v_obj.object_name);
     END LOOP;
 
+    -- Remove entries from pg_catalog.pg_dist_tenant_schema for all dropped tenant schemas.
+    -- Also delete the corresponding colocation group from pg_catalog.pg_dist_colocation.
+    --
+    -- Although normally we automatically delete the colocation groups when they become empty,
+    -- we don't do so for the colocation groups that are created for tenant schemas. For this
+    -- reason, here we need to delete the colocation group when the tenant schema is dropped.
+    FOR v_obj IN SELECT * FROM pg_event_trigger_dropped_objects() WHERE object_type = 'schema'
+    LOOP
+        IF EXISTS (SELECT 1 FROM pg_catalog.pg_dist_tenant_schema WHERE schemaid = v_obj.objid)
```

Comment:
this could be a join in the query above

Reply:
(alternatively, could be folded into loop below)

Reply:
(will do soon 👍)

Reply:
done